pkg/controller/disruption/disruption.go

```diff
@@ -173,94 +175,84 @@ func (dc *DisruptionController) finders() []podControllerFinder {
 		dc.getPodStatefulSets}
 }
 
+var (
+	controllerKindRS  = v1beta1.SchemeGroupVersion.WithKind("ReplicaSet")
+	controllerKindSS  = apps.SchemeGroupVersion.WithKind("StatefulSet")
+	controllerKindRC  = v1.SchemeGroupVersion.WithKind("ReplicationController")
+	controllerKindDep = v1beta1.SchemeGroupVersion.WithKind("Deployment")
+)
+
 // getPodReplicaSets finds replicasets which have no matching deployments.
 func (dc *DisruptionController) getPodReplicaSets(pod *v1.Pod) ([]controllerAndScale, error) {
-	cas := []controllerAndScale{}
-	rss, err := dc.rsLister.GetPodReplicaSets(pod)
-	// GetPodReplicaSets returns an error only if no ReplicaSets are found.  We
-	// don't return that as an error to the caller.
-	if err != nil {
-		return cas, nil
-	}
-	controllerScale := map[types.UID]int32{}
-	for _, rs := range rss {
-		// GetDeploymentsForReplicaSet returns an error only if no matching
-		// deployments are found.
-		_, err := dc.dLister.GetDeploymentsForReplicaSet(rs)
-		if err == nil { // A deployment was found, so this finder will not count this RS.
-			continue
+	var casSlice []controllerAndScale
+	if controllerRef := controller.GetControllerOf(pod); controllerRef != nil {
+		if controllerRef.Kind == controllerKindRS.Kind {
+			rs, err := dc.rsLister.ReplicaSets(pod.Namespace).Get(controllerRef.Name)
+			if err != nil {
+				return nil, nil
+			}
+			controllerRef := controller.GetControllerOf(rs)
+			if controllerRef == nil || controllerRef.Kind != controllerKindDep.Kind {
+				casSlice = append(casSlice, controllerAndScale{rs.UID, *(rs.Spec.Replicas)})
```

Comment:
Go style suggestion: This line is the end of the "happy path" (everything went as expected - we found an RS). Ideally it should be as unindented as possible, and unexpected things should result in early `return` statements. This way the "happy path" flows straight down rather than becoming more deeply nested at each step.

As an example, this function could be structured like:

```go
...
if controllerRef == nil {
  return nil, nil
}
if controllerRef.Kind != controllerKindRS.Kind {
  return nil, nil
}
rs, err := ...
if err != nil {
  // The only possible error is NotFound, which is ok here.
  return nil, nil
}
if rs.UID != controllerRef.UID {
  return nil, nil
}
...
if controllerRef != nil && controllerRef.Kind == controllerKindDep.Kind {
  // Skip RS if it's controlled by a Deployment.
  return nil, nil
}
return []controllerAndScale{...}
```

Notice there's no nesting and everything flows straight down.